packages/vscode-dependencies-validation/src/autofix/eventUtil.ts

```diff
@@ -0,0 +1,23 @@
+import type { Uri } from "vscode";
+import { isAutoFixEnabled } from "./configuration";
+import {
+  clearDiagnostics,
+  findAndFixDepsIssues,
+  isNotInNodeModules,
+} from "../util";
+import { VscodeFileEventConfig, VscodeWorkspace } from "../vscodeTypes";
+
+export async function handlePackageJsonEvent(
+  uri: Uri,
+  vscodeConfig: VscodeFileEventConfig
+): Promise<void> {
+  const { workspace, diagnosticCollection, outputChannel } = vscodeConfig;
+  if (canBeFixed(workspace, uri)) {
+    await findAndFixDepsIssues(uri, outputChannel);
+    clearDiagnostics(diagnosticCollection, uri);
```

Comment:
I think there is an hidden assumption here that `findAndFixDepsIssues` always succeeded and that the 
`clearDiagnostics` is always safe/correct to execute.

Reply:
No, it will not reach clearDiagnostics if findAndFixDepsIssues fails